### ISSUE TITLE
[Feature]/RFC - Continue x/y/z script on individual image failure.

### ISSUE THREAD

**Issue by devdave** [vladmandic/automatic#1122]
### Feature description

If there is an error in a large job (+1 hour of processing), the script will stop the entire batch job.   It would be nice to have a checkbox option to just keep going on cell error.

I've been looking at the x/y/z script and it seems possible but I might have the wrong spot in the script python file (process_cell inside xyz_script which is called by run).

I can try and fork the script and do some quick hacking to see if it's possible but wanted to know if 1. you are already working on something similar 2. if I do make a clean PR do you think you would want this feature?

### Version Platform Description

Latest version as I do daily pull and merge with your repo.

**Comment by vladmandic** [comment]
it all depends on error itself - some can be captured harmlessly and others do cause internal inconsistencies.
and i'm not sure which type of an error would occur once during long job, but not on any other jobs?

**Comment by devdave** [comment]
tl;dr Catch, report, but continue despite a Pytorch/tensor flow originating exceptions.   Maybe have some of "bad checkpoint" set so the script can skip over additional tasks related to that model?

My normal workflow is to come up with a prompt, set a ControlNet pose, and then use x/y/z script to test that construct against 50-60 models across the spectrum.

A good example of where problems come up is with models like this one (safe for work despite the name- https://civitai.com/models/2107/fkingscifiv2) so it's perfect for testing.     It has some sort of geometry problem that clashes with the other models because it does not use the standard 512x512 image sizing.  That or it has a problem with both the size and the use of ControlNet's pose?    Unfortunately, it isn't the only checkpoint model with problems.     

Having an hour-long batch job fail in the first 10-15 minutes can be a bit frustrating, especially if triggering the x/y/z job to run and then stepping away from the workstation.

**Comment by vladmandic** [comment]
that is a good example, but i'd need to test that exact scenario - i don't want to add global try/except without actually knowing what it catches :)

**Comment by TeutonJon78** [comment]
One error I ran into was doing a large x/y/z, the final z image was beyond the "megapixel limit" in settings, so the script stalled. But it was generating the z image before the x/y ones, so I didn't get any of the grids from the run, just all the individual images.

**Comment by arcum42** [comment]
Actually, an easy example of individual image failure on an x/y/z script is to use a lora in your prompt, then have the script run it for multiple models, and include a model the lora doesn't work with. (i.e.: mix 2.x and 1.x models in the list of models you give the script.)

**Comment by vladmandic** [comment]
That's a good example

**Comment by TeutonJon78** [comment]
To add to my previous failure scenario, another common failure would be if the GPU runs out of vRAM. The single generation fails and then the script fails out as well.  Then it doesn't end up generating any of the grids for the run, even for all the completed images.

**Comment by vladmandic** [comment]
running into any scenario where torch gets into inconsistent state is not really recoverable - and that includes oom.

**Comment by vladmandic** [comment]
recoverable failures are now caught in dev branch.
unrecoverable failures, well, cannot be caught as they lead into inconsistent state.
